### PR TITLE
feat: proximity placement group support

### PR DESCRIPTION
**Pull request by kkmsft**
**Reason for Change**:

Support Proximity Placement Group (PPG) for node profile. PPG allows co-locating Azure resources. In this case we can use PPG to co-locate the VMs in an agentpool or in the master profile. More details on PPG can be found at: https://azure.microsoft.com/en-us/blog/introducing-proximity-placement-groups/

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by kkmsft** [comment]
cc: @andyzhangx @tilnl

File: docs/topics/clusterdefinitions.md [hunk 1]
```diff
@@ -732,7 +732,7 @@ A cluster can have 0 to 12 agent pool profiles. Agent Pool Profiles are used for
...
-
+| proximityPlacementGroup          | no                        | Specifies the resource id of the Proximity Placement Group (PPG) to use for this agentpool.  Please find more details about PPG in this [Azure blog](https://azure.microsoft.com/en-us/blog/introducing-proximity-placement-groups). Note that the PPG should be created in advance. The following [Azure CLI documentation](https://docs.microsoft.com/en-us/cli/azure/ppg?view=azure-cli-latest#az-ppg-create) explains how to create a PPG.                                                                                                                                                              |
```
Comment (andyzhangx) [inline docs/topics/clusterdefinitions.md:735]: `proximityPlacementGroupID` is better IMO

Comment (jackfrancis) [reply docs/topics/clusterdefinitions.md:735]: Based on the way the actual ARM spec is composed I agree with Andy:

E.g.,

```
    if masterProfile.ProximityPlacementGroup != "" {
 		ppg := masterProfile.ProximityPlacementGroup
 		vmProperties.ProximityPlacementGroup = &compute.SubResource{
 			ID: &ppg,
 		}
 	}
```

My only other thought: is there any chance that additional `ProximityPlacementGroup` properties will be exposed on the api model? Should we make the api model `ProximityPlacementGroup` config interface a struct? If not, then let's call it `ProximityPlacementGroupID`.

File: pkg/engine/virtualmachinescalesets.go [hunk 2]
```diff
@@ -441,6 +447,13 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 		vmssProperties.PlatformFaultDomainCount = to.Int32Ptr(int32(*profile.PlatformFaultDomainCount))
 	}
 
+	if profile.ProximityPlacementGroup != "" {
+		ppg := profile.ProximityPlacementGroup
+		vmssProperties.ProximityPlacementGroup = &compute.SubResource{
+			ID: to.StringPtr(ppg),
```
Comment (jackfrancis) [inline pkg/engine/virtualmachinescalesets.go:453]: Let's standardize to one usage pattern here. We can probably get rid of the temporary `ppg` variable and just use:

`ID: to.StringPtr(profile.ProximityPlacementGroup),`

**Comment by jackfrancis** [comment]
Added some comments, generally this looks good, thanks @kkmsft!

File: docs/topics/clusterdefinitions.md (hunk 1 above)
Comment (kkmsft) [reply docs/topics/clusterdefinitions.md:735]: @andyzhangx , @jackfrancis - proximityPlacementGroupID makes sense. Will update the PR.

@jackfrancis - as now don't anticipate any further properties being exposed.

File: pkg/engine/virtualmachinescalesets.go (hunk 2 above)
Comment (kkmsft) [reply pkg/engine/virtualmachinescalesets.go:453]: sure. will update the PR.

**Comment by kkmsft** [comment]
@jackfrancis : Have addressed your and Andy's comments. PTAL.

**Comment by tilnl** [comment]
LGTM

**Comment by andyzhangx** [comment]
@kkmsft will you add the ppg id format validation?
https://github.com/Azure/aks-engine/pull/3056#pullrequestreview-391728517

**Comment by kkmsft** [comment]
@andyzhangx : Updated with the validate checks.

**Pull request merged**